code/modules/spells/spell.dm

```diff
@@ -368,6 +368,12 @@
 		return FALSE
 
 	var/mob/living/living_owner = owner
+	switch(SEND_SIGNAL(owner, COMSIG_MOB_TRY_INVOKE_SPELL, src, feedback))
+		if(SPELL_INVOCATION_FAIL)
+			return FALSE
+		if(SPELL_INVOCATION_SUCCESS)
+			return TRUE // skips all of the following checks
```

Comment:
```suggestion
	var/invoke_sig_return = SEND_SIGNAL(owner, COMSIG_MOB_TRY_INVOKE_SPELL, src, feedback)
	if(invoke_sig_return & SPELL_INVOCATION_ALWAYS_SUCCEED)
		return TRUE // skips all of the following checks
	if(invoke_sig_return & SPELL_INVOCATION_FAIL)
		return FALSE
```
Switch is not good since it's a bitflag, multiple signals might return multiple results.